src/resolvers/git.cr

```diff
@@ -46,12 +46,16 @@ module Shards
     def initialize(@commit : String)
     end
 
+    def =~(other : GitCommitRef)
+      commit.size >= 7 && other.commit.starts_with?(commit)
+    end
+
     def to_git_ref
       @commit
     end
 
-    def to_s(io)
-      io << "commit " << @commit[0...7]
+    def to_s(io, *, commit_length = nil)
```

Comment:
I don't see where this `commit_length` argument is in use

Reply:
I originally wanted to use that to display different lengths depending on context, but decided that's too complex for this fix PR.

Reply:
Please, check my comment: https://github.com/crystal-lang/shards/pull/386#issuecomment-626090079

Actually I think the right way to solve this is raise an error with a better message at `latest_version_for_ref` when there is no available matches, (instead of returning `nil`). That message should contain the exact input read from the `shard.yml`. And leave here the abbreviated commit. At that point it will be used only to show conflicts with other dependencies, but already knowing the commit exists.